core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -375,7 +491,19 @@ public int size() {
    * @param value the field value
    * @return the set of objects with the specified field value
    */
-  private ConcurrentHashSet<T> getByFieldInternal(FieldIndex<T> index, Object value) {
-    return mIndexMap.get(index).get(value);
+  private T getByFieldInternalUniq(FieldIndex<T> index, Object value) {
```

Comment:
If we have `UniqueFeildIndex`, we could get rid of the following check and being type-safe meanwhile